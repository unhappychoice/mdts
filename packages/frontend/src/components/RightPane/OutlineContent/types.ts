export interface OutlineItem {
  id: string;
  level: number;
  content: string;
}
