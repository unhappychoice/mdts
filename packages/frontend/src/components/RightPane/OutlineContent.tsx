import { CircularProgress, List, ListItem, ListItemText, Box } from '@mui/material';
import React from 'react';

interface OutlineItem {
  id: string;
  level: number;
  content: string;
}

interface OutlineContentProps {
  outline: OutlineItem[];
  loading: boolean;
  error: string | null;
  onItemClick: (id: string) => void;
}

const OutlineContent: React.FC<OutlineContentProps> = ({ outline, loading, error, onItemClick }) => {
  if (loading) {
    return (
      <Box sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: 'calc(100vh - 132px)' }}>
        <CircularProgress />
      </Box>
    );
  }

  if (error) {
    return <p>Error: {error}</p>;
  }

  return (
    <List dense sx={{ overflowY: 'auto', height: 'calc(100vh - 132px)' }}>
      {outline.map((item, index) => (
        <ListItem button key={`${item.id}-${index}`} sx={{ pl: item.level * 2 }} onClick={() => onItemClick(item.id)}>
          <ListItemText
            primary={item.content}
            sx={{
              overflow: 'hidden',
              whiteSpace: 'nowrap',
              textOverflow: 'ellipsis',
            }}
          />
        </ListItem>
      ))}
    </List>
  );
};

export default OutlineContent;
