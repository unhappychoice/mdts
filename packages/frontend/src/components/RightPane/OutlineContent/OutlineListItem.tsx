import { ListItem, ListItemText } from '@mui/material';
import React, { useCallback } from 'react';
import { OutlineItem } from './types';

interface OutlineListItemProps {
  item: OutlineItem;
  onItemClick: (id: string) => void;
}

export const OutlineListItem: React.FC<OutlineListItemProps> = ({ item, onItemClick }) => {
  const handleClick = useCallback(() => {
    onItemClick(item.id);
  }, [onItemClick, item.id]);

  return (
    <ListItem button key={item.id} sx={{ pl: item.level * 2 }} onClick={handleClick}>
      <ListItemText primary={item.content} sx={{ overflow: 'hidden', whiteSpace: 'nowrap', textOverflow: 'ellipsis' }} />
    </ListItem>
  );
};
