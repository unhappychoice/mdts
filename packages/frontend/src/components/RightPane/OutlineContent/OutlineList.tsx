import { List } from '@mui/material';
import React from 'react';
import { OutlineListItem } from './OutlineListItem';
import { OutlineItem } from './types';

interface OutlineListProps {
  outline: OutlineItem[];
  onItemClick: (id: string) => void;
}

export const OutlineList: React.FC<OutlineListProps> = ({ outline, onItemClick }) => (
  <List dense className="custom-scrollbar" sx={{ overflowY: 'auto', height: 'calc(100vh - 132px)', px: 2 }}>
    {outline.map((item) => (
      <OutlineListItem key={item.id} item={item} onItemClick={onItemClick} />
    ))}
  </List>
);
