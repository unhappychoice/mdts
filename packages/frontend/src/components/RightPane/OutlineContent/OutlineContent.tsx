import React from 'react';
import ErrorView from '../../ErrorView';
import { LoadingIndicator } from './LoadingIndicator';
import { OutlineList } from './OutlineList';
import { OutlineItem } from './types';

interface OutlineContentProps {
  outline: OutlineItem[];
  loading: boolean;
  error: string | null;
  onItemClick: (id: string) => void;
}

const OutlineContent: React.FC<OutlineContentProps> = ({ outline, loading, error, onItemClick }) => {
  if (loading) {
    return <LoadingIndicator />;
  }

  if (error) {
    return <ErrorView error={error} />;
  }

  return <OutlineList outline={outline} onItemClick={onItemClick} />;
};

export default OutlineContent;
