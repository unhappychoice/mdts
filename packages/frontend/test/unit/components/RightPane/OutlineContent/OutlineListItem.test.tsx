import { render } from '@testing-library/react';
import React from 'react';
import { OutlineListItem } from '../../../../../src/components/RightPane/OutlineContent/OutlineListItem';

describe('OutlineListItem', () => {
  it('should render correctly', () => {
    const item = { id: '1', level: 1, content: 'Heading 1' };
    const onItemClick = jest.fn();

    const { asFragment } = render(
      <OutlineListItem item={item} onItemClick={onItemClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
