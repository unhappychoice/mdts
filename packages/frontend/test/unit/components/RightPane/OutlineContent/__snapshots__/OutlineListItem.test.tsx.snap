// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`OutlineListItem should render correctly 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-66b8d4-MuiListItem-root"
  >
    <div
      class="MuiListItemText-root css-6qv7tz-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1px98du-MuiTypography-root"
      >
        Heading 1
      </span>
    </div>
  </li>
</DocumentFragment>
`;
